src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -1356,7 +1313,63 @@ private Autogenerated.StateItem ToAutogeneratedStateItem(StateTransactionRequest
             return new UnlockResponse(GetUnLockStatus(response.Status));
         }
 
-        #endregion 
+        #endregion
+
+        #region Dapr Sidecar Methods
+
+        /// <inheritdoc/>
+        public override async Task<bool> CheckHealthAsync(CancellationToken cancellationToken = default)
+        {
+            var path = "/v1.0/healthz";
+            var request = new HttpRequestMessage(HttpMethod.Get, new Uri(this.httpEndpoint, path));
+            try
+            {
+                var response = await this.httpClient.SendAsync(request, cancellationToken);
+                return response.IsSuccessStatusCode;
+            }
+            catch (HttpRequestException)
+            {
+                return false;
+            }
+        }
+
+        /// <inheritdoc/>
+        public override async Task<bool> CheckOutboundHealthAsync(CancellationToken cancellationToken = default)
+        {
+            var path = "/v1.0/healthz/outbound";
+            var request = new HttpRequestMessage(HttpMethod.Get, new Uri(this.httpEndpoint, path));
+            try
+            {
+                var response = await this.httpClient.SendAsync(request, cancellationToken);
```

Comment:
nitpicking:

This is a really lightweight request but given that you don't even need the request content you can ignore it by adding `HttpCompletionOption.ResponseHeadersRead`
```suggestion
                using var response = await this.httpClient.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken);
```